setup.py

```diff
@@ -77,8 +77,8 @@
     f"jax=={jax_version}",
     f"jaxlib=={jax_version}",
     "tomlkit; python_version < '3.11'",
-    "scipy<1.13",
-    "numpy<2",
+    "scipy==1.13",
```

Comment:
I'm find with this change, but I know you tried to unpin scipy. 

It looks like[ pennylane doesn't pin its scipy version](https://github.com/PennyLaneAI/pennylane/blob/68fd307ded65216231dbea6b309709103515c165/setup.py#L25). Not necessary for this PR to be merged, but do you remember why we couldn't move to the latest scipy?

Reply:
Yeah you are right. although they pin it in the [requirement-ci.txt](https://github.com/PennyLaneAI/pennylane/pull/6061/files#diff-881c8173c25e0340286efa57853f9d953b9e3573ecb7ec2c07a90ecfd39ea0d2). 
When using scipy 1.14 I get this error: 
```
********************
FAIL: Frontend Tests :: test_callback.py (13 of 27)
******************** TEST 'Frontend Tests :: test_callback.py' FAILED ********************
Exit Code: 2

Command Output (stderr):
--
RUN: at line 17: /Users/mehrdad.malek/catalyst/venv/bin/python3 /Users/mehrdad.malek/catalyst/frontend/test/lit/test_callback.py | FileCheck /Users/mehrdad.malek/catalyst/frontend/test/lit/test_callback.py
+ /Users/mehrdad.malek/catalyst/venv/bin/python3 /Users/mehrdad.malek/catalyst/frontend/test/lit/test_callback.py
+ FileCheck /Users/mehrdad.malek/catalyst/frontend/test/lit/test_callback.py
/Users/mehrdad.malek/catalyst/venv/lib/python3.12/site-packages/pennylane/__init__.py:184: PennyLaneDeprecationWarning: QubitDevice will no longer be accessible top level. Please access  the class as pennylane.devices.QubitDevice
  warn(
Traceback (most recent call last):
  File "/Users/mehrdad.malek/catalyst/frontend/test/lit/test_callback.py", line 30, in <module>
    @qml.qjit
     ^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/venv/lib/python3.12/site-packages/pennylane/compiler/qjit_api.py", line 308, in qjit
    return qjit_loader(fn=fn, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/venv/lib/python3.12/site-packages/pennylane/logging/decorators.py", line 61, in wrapper_entry
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/frontend/catalyst/jit.py", line 377, in qjit
    return QJIT(fn, CompileOptions(**kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/venv/lib/python3.12/site-packages/pennylane/logging/decorators.py", line 65, in wrapper_exit
    output = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/frontend/catalyst/jit.py", line 445, in __init__
    self.aot_compile()
  File "/Users/mehrdad.malek/catalyst/venv/lib/python3.12/site-packages/pennylane/logging/decorators.py", line 61, in wrapper_entry
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/frontend/catalyst/jit.py", line 491, in aot_compile
    self.compiled_function, self.qir = self.compile()
                                       ^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/frontend/catalyst/debug/instruments.py", line 143, in wrapper
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/venv/lib/python3.12/site-packages/pennylane/logging/decorators.py", line 61, in wrapper_entry
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/frontend/catalyst/jit.py", line 672, in compile
    shared_object, llvm_ir, _ = self.compiler.run(self.mlir_module, self.workspace)
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/venv/lib/python3.12/site-packages/pennylane/logging/decorators.py", line 61, in wrapper_entry
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/frontend/catalyst/compiler.py", line 610, in run
    return self.run_from_ir(
           ^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/venv/lib/python3.12/site-packages/pennylane/logging/decorators.py", line 61, in wrapper_entry
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/frontend/catalyst/compiler.py", line 586, in run_from_ir
    output = LinkerDriver.run(filename, options=self.options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/venv/lib/python3.12/site-packages/pennylane/logging/decorators.py", line 61, in wrapper_entry
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/frontend/catalyst/compiler.py", line 511, in run
    flags = LinkerDriver.get_default_flags(options)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/venv/lib/python3.12/site-packages/pennylane/logging/decorators.py", line 61, in wrapper_entry
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mehrdad.malek/catalyst/frontend/catalyst/compiler.py", line 394, in get_default_flags
    raise CompileError(
catalyst.utils.exceptions.CompileError: Unable to find OpenBLAS library at "/Users/mehrdad.malek/catalyst/venv/lib/python3.12/site-packages/scipy/.dylibs/libopenblas*.dylib". Please ensure that SciPy is installed and available via pip.
FileCheck error: '<stdin>' is empty.
FileCheck command line:  FileCheck /Users/mehrdad.malek/catalyst/frontend/test/lit/test_callback.py
```

I have not looked into it, but there is some problem loading openblas library.

Reply:
Thanks! I think this is ok for now as this is not related to numpy 2.0, but something to look into later.